agent/hcp/telemetry_provider.go

```diff
@@ -183,3 +245,54 @@ func (h *hcpProviderImpl) IsDisabled() bool {
 
 	return h.cfg.disabled
 }
+
+// updateHTTPConfig updates the HTTP configuration values that rely on the HCP configuration.
+func (h *hcpProviderImpl) updateHTTPConfig(cfg config.CloudConfigurer) error {
+	h.httpCfgRW.Lock()
+	defer h.httpCfgRW.Unlock()
+
+	if cfg == nil {
+		return errors.New("must provide valid HCP configuration")
+	}
+
+	// Update headers
+	r, err := cfg.Resource()
+	if err != nil {
+		return fmt.Errorf("failed set telemetry client headers: %v", err)
+	}
+	header := make(http.Header)
+	header.Set("content-type", "application/x-protobuf")
+	header.Set("x-hcp-resource-id", r.String())
+	header.Set("x-channel", fmt.Sprintf("consul/%s", version.GetHumanVersion()))
+	h.httpCfg.header = &header
+
+	// Update HTTP client
+	hcpCfg, err := cfg.HCPConfig()
+	if err != nil {
+		return fmt.Errorf("failed to configure telemetry HTTP client: %v", err)
+	}
+	h.httpCfg.client = client.NewHTTPClient(
+		hcpCfg.APITLSConfig(),
+		hcpCfg,
+		h.logger.Named("hcp_telemetry_client"))
+
+	return nil
+}
+
+// GetHeader acquires a read lock to return the HTTP request headers needed
+// to export metrics.
+func (h *hcpProviderImpl) GetHeader() *http.Header {
```

Comment:
```
// GetHeader acquires a read lock to return the HTTP request headers needed
// to export metrics.
func (h *hcpProviderImpl) GetHeader() http.Header {
	h.httpCfgRW.RLock()
	defer h.httpCfgRW.RUnlock()

	return h.httpCfg.header.Clone()
}
```
- Clones header since it's a map (in case it gets written to at same time)
- Removes need for dereference in `metrics_client.go`

Reply:
Went ahead and made this change since it makes it more thread safe!